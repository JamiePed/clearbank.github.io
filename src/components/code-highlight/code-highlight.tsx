import React from 'react'
import Highlight, { defaultProps } from 'prism-react-renderer'
import vsDark from 'prism-react-renderer/themes/vsDark'
import * as Types from './code-highlight.types'

const CodeHighlight: React.FunctionComponent<Types.CodeHighlightProps> = ({
  code,
  language
}: Types.CodeHighlightProps) => (
  <Highlight {...defaultProps} code={code} language={language} theme={vsDark}>
    {({ className, style, tokens, getLineProps, getTokenProps }) => (
      <pre className={className} style={style}>
        {tokens.map((line, i) => (
          <div key={i} {...getLineProps({ line, key: i })}>
            {line.map((token, key) => (
              <span key={key} {...getTokenProps({ token, key })} />
            ))}
          </div>
        ))}
      </pre>
    )}
  </Highlight>
)

export default CodeHighlight
