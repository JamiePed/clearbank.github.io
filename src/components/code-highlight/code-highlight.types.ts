export interface CodeHighlightProps {
  code: string
  language?: string
}
