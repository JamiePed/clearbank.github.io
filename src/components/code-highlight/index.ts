export { default } from './code-highlight'
