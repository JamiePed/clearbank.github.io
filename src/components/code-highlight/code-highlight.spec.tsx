import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import 'jest-styled-components'

import Component from './code-highlight'

import * as Types from './code-highlight.types'

let component: ReturnType<typeof render> = null
let fragment: DocumentFragment = null
let root: HTMLElement = null

const defaultProps: Types.CodeHighlightProps = {
  code: 'hello world',
  language: 'turbo-pascal'
}

beforeEach(() => {
  component = render(<Component {...defaultProps} children='children' />)

  fragment = component.asFragment()
  root = component.container.firstChild
})

afterEach(() => {
  fragment = null
  root = null
})

describe('Component', () => {
  test('matches snapshot', () => {
    expect(root).toMatchSnapshot()
  })

  test('has attributes from props', () => {
    expect(root.classList.contains(`language-${defaultProps.language}`)).toBe(
      true
    )
  })

  test('contains at least one pre-element', () => {
    expect(fragment.querySelectorAll('pre').length).toBeGreaterThanOrEqual(1)
  })

  test('is not empty', () => {
    expect(fragment.textContent.length).toBeTruthy()
  })
})
