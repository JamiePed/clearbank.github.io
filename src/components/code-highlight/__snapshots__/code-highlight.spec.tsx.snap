// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component matches snapshot 1`] = `
<pre
  class="prism-code language-turbo-pascal"
  style="color: rgb(156, 220, 254); background-color: rgb(30, 30, 30);"
>
  <div
    class="token-line"
    style="color: rgb(156, 220, 254);"
  >
    <span
      class="token plain"
    >
      hello world
    </span>
  </div>
</pre>
`;
